packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -23,6 +24,42 @@ import {
 } from './InternalHelpers';
 import { IHttpNegotiationResult, NegotiatePartialOptions, NegotiationOptions } from './types';
 
+const warnMsg = (contentTypes: string[]) => `Unable to find content for ${contentTypes}`;
+
+const optionallyGetPayloadlessResponse = (
+  response: IHttpOperationResponse,
+  headers: IHttpHeaderParam[],
+  mediaTypes: string[],
+): Option.Option<IHttpNegotiationResult> => {
+  return pipe(
+    Option.fromNullable(response.contents),
+    Option.map(contents =>
+      contents.filter(c => {
+        return !(c.mediaType === '*/*');
+      }),
+    ),
+    Option.map(contents =>
+      contents.filter(c => {
+        return !c.schema && !(c.examples || []).length;
+      }),
+    ),
+    Option.chain(responses => {
+      return Option.fromPredicate((noContentResponses: { length: number }) => {
+        const noAcceptAndNoContentResponses =
+          !noContentResponses.length && !mediaTypes.filter((mt: string) => !mt.includes('*/*')).length;
```

Comment:
I really think we should put a comment on this condition, I still struggle to understand it correctly.

Also, with includes `Accept: application/json,*/*` will match. Is this intentional? I guess so (if the client accepts */* then an empty response is viable, right?)